nautobot/dcim/forms.py

```diff
@@ -696,21 +696,21 @@ def clean(self):
         cleaned_data = self.cleaned_data
         site = cleaned_data.get("site")
 
-        if self.instance:
+        if self.instance and site != self.instance.site:
```

Comment:
This is causing issues in the unit tests, perhaps we need to check if `self.instance.site` exists as well?

Reply:
Should be fixed now, was missing a `self.instance.present_in_database` check